src/main/io/gps.c

```diff
@@ -2637,6 +2639,30 @@ void GPS_calculateDistanceAndDirectionToHome(void)
     }
 }
 
+void GPS_calculateRelativePositionEarthGround(int32_t *homeLat, int32_t *homeLon, int32_t *targetLat, int32_t *targetLon, gpsRelativePos_t *gpsrelativePos) {
+{
+    if (STATE(GPS_FIX_HOME)) {
+        // Umrechnung von Grad in Bogenmaß
+        float homeLatRad = (*homeLat / 10000000.0) * M_PI / 180.0;
+        float homeLonRad = (*homeLon / 10000000.0) * M_PI / 180.0;
+        float targetLatRad = (*targetLat / 10000000.0) * M_PI / 180.0;
+        float targetLonRad = (*targetLon / 10000000.0) * M_PI / 180.0;
+
+        // Berechnung der relativen Position
+        float deltaLat = targetLatRad - homeLatRad;
+        float deltaLon = targetLonRad - homeLonRad;
+
+        // Berechnung der Entfernung in Metern
+        //float distance = GPS_EARTH_RADIUS * sqrt(deltaLat * deltaLat + deltaLon * deltaLon);
+
+        gpsrelativePos->lat = (uint16_t)(deltaLat * GPS_EARTH_RADIUS);
+        gpsrelativePos->lon = (uint16_t)(deltaLon * GPS_EARTH_RADIUS);
```

Comment:
This won't work - meridians get closer as you go from equator to pole

Reply:
Yes thats why I do not understand why you want to go to earth coordinats. I think we get bitter failere of value then with my current implementation in this PR.

Reply:
GPS_distance_cm_bearing does the calculation and does it correctly (it can be prettier, though), so the code can be reused to get cartesian distance (in cm)

As I explained before - Earth frame is used in BF code. It may be a bit misleading (it is not ECEF). Ground frame would be a bit better in this context.
It is cartesian frame fixed at given location, with XY tangent to Earth surface. Due to  historical heritage, NWU system is closest to what is used in BF.